gadgets/top_file/test/top_file_test.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	gadgettesting "github.com/inspektor-gadget/inspektor-gadget/gadgets/testing"
+	igtesting "github.com/inspektor-gadget/inspektor-gadget/pkg/testing"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/containers"
+	igrunner "github.com/inspektor-gadget/inspektor-gadget/pkg/testing/ig"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/match"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/utils"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+
+type topFileEntry struct {
+	eventtypes.CommonData
+
+	MountNsID uint64 `json:"mntns_id"`
+
+	Pid  uint32 `json:"pid"`
+	Tid  uint32 `json:"tid"`
+	Comm string `json:"comm"`
+
+	Reads      uint64 `json:"reads"`
+	Writes     uint64 `json:"writes"`
+	ReadBytes  uint64 `json:"rbytes"`
+	WriteBytes uint64 `json:"wbytes"`
+
+	FileType  string `json:"t"`
+	FileName  string `json:"file"`
+	FileInode uint64 `json:"inode"`
+
+	// Device might be 0, so we can't use NormalizedInt and NormalizeInt()
+	// TODO: Support checking for the presence of a field, even if it's 0
```

Comment:
Had the same issue in https://github.com/inspektor-gadget/inspektor-gadget/pull/3115#discussion_r1667073483